protofsm/state_machine.go

```diff
@@ -145,7 +145,7 @@ type StateMachine[Event any, Env Environment] struct {
 	// query the internal state machine state.
 	stateQuery chan stateQuery[Event, Env]
 
-	wg   fn.GoroutineManager
+	wg   *fn.GoroutineManager
```

Comment:
IMHO, this change isn't necessary. `go vet` already ensures that non-copiable objects like mutexes or wait groups are not copied. By keeping the previous version, we avoid adding unnecessary indirection to operations involving the GoroutineManager.

Additionally, I suggest renaming `wg` to `gm` to avoid any confusion with a standard wait group.

Reply:
cool, did both